### DISCUSSION TITLE
How to debug front-end code with vscode?

### DISCUSSION THREAD

**Discussion by davidxwwang** [eclipse-emfcloud/emfcloud#135] [Coffee Editor] [answered]
How to debug front-end code with vscode?

**Comment by ndoschek** [comment, marked as answer]
Hi @davidxwwang!
You can debug frontend code in VSCode by starting the launch configuration `Launch Browser Frontend` in the 'Run and Debug' view. This will open a separate Chrome instance via the VSCode extension [Debugger for Chrome](https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome) for example (under the precondition that the servers were already started).

Here you can see it in action:
![discussions-browser-debugging](https://user-images.githubusercontent.com/47414588/157869779-1846c708-c05a-4324-b092-8aed68fb7c56.gif)

You can also check out the VSCode docs about [Browser debugging](https://code.visualstudio.com/docs/nodejs/browser-debugging).

Hope that helps and best wishes,
Nina
